src/lib/lottoLib.js

```diff
@@ -0,0 +1,35 @@
+import {
+  LOTTO_BONUS_NUMBER_TYPE,
+  LOTTO_PRICE_TYPE,
+  LOTTO_WINNER_NUMBERS_TYPE,
+} from "../constants/lotto.js";
+
+const isComma = (input) => {
+  if (input.includes(",")) {
+    return true;
+  }
+
+  return false;
+};
+
+const isNumber = (input) => {
+  return /^-?\d+(\.\d+)?$/.test(input);
+};
+
+export const checkByType = (type, input) => {
+  if (
+    (type === LOTTO_PRICE_TYPE || type === LOTTO_BONUS_NUMBER_TYPE) &&
+    !isComma(input)
+  ) {
+    return isNumber(input);
+  }
+
+  if (type === LOTTO_WINNER_NUMBERS_TYPE && isComma(input)) {
+    const arrayNumber = input.replace(/ /g, "").split(",");
+    const filterIsNumber = arrayNumber.filter((el) => isNumber(el)).length;
+
+    return arrayNumber.length === filterIsNumber;
+  }
+
+  return false;
+};
```

Comment:
지금 3가지 인풋 타입에 대한 체크를 하고 있는데, 이걸 이런 방식으로 표현해줄 수 있도록 함수를 쪼개주면 어떨까요!?

```suggestion
export const checkByType = createTypeCheck({
  LOTTO_PRICE_TYPE: isPriceType,
  LOTTO_BONUS_NUMBER_TYPE: isBonusNumberType,
  LOTTO_WINNER_NUMBERS_TYPE: isWinnerNumbersType,
})
```

무조건 이렇게 해야돼! 라기 보단, 함수 하나가 여러 개의 일을 하고 있는데 이를 밖에서 조합할 수 있도록 만들어주면 좋을 것 같아요!